server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -294,22 +295,23 @@ public CompletionStage<Result> updateFile(
                     originalFileName.get());
               }
 
-              updateFileRecord(key.get(), originalFileName);
-              return applicantService.stageAndUpdateIfValid(
```

Comment:
note to reviewer: the previous behavior made an async call to insert the file record but did not handle the resulting promise, which made silent failure possible if the code to insert the record failed for some reason

Reply:
@bion Would it have been possible that a file made it's way into S3, but didn't get recorded in the database? I can pull metadata from S3 if we need to check or rebuild missing records.